### PR TITLE
Back button support

### PR DESCRIPTION
**Pull request by svenjens**
This PR takes care of supporting the browser back button. 

Let's discuss: Should the back button also save your current answer in the question?

I think going back should always discard all the choices you made on that page.

### CODE REVIEW

File: apps/client/src/components/Nav.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { NavStyle } from "./NavStyle";
 import { useRouteMatch } from "react-router-dom";
 import { routeConfig } from "../routes";
 import Context from "../context";
+import { useHistory } from "react-router-dom";
```
Comment (afjlambert) [inline apps/client/src/components/Nav.js:11]: why?

File: apps/client/src/pages/QuestionsPage.js [hunk 2]
```diff
@@ -15,19 +15,23 @@ const QuestionsPage = ({ topic, checker }) => {
...
-          slug: topic.slug,
-          question: currSlug,
-        })}
-      />
+  if (!questionSlug) {
+    history.push(
+      geturl(routes.questions, {
+        slug: topic.slug,
+        question: currSlug,
+      })
```
Comment (afjlambert) [inline apps/client/src/pages/QuestionsPage.js:34]: Why do you want to change the `if` condition? Please also note the comment above it.
Also, why did you move from `Redirect` to `history.push`? Is this better?

File: apps/client/src/components/Nav.js (hunk 1 above)
Comment (svenjens) [reply apps/client/src/components/Nav.js]: history.push add a route to the history stack. 
So you can use it to navigate back by using the native button. 

Redirect is like a 3xx server redirect.
